packages/mermaid/src/diagrams/timeline/svgDraw.js

```diff
@@ -523,23 +524,40 @@ export const drawNode = function (elem, node, fullSection, conf) {
 
   // Create the wrapped text element
   const textElem = nodeElem.append('g');
+  const description = node.descr.replace(/(<br\s*\/?>)/g, '\n');
+  const htmlLabels = conf.timeline.htmlLabels;
+
+  createText(textElem, description, {
+    useHtmlLabels: htmlLabels,
+    width: node.width,
+    classes: 'timeline-node-label',
+  });
+
+  if (!htmlLabels) {
+    textElem
+      .attr('dy', '1em')
+      .attr('alignment-baseline', 'middle')
+      .attr('dominant-baseline', 'middle')
+      .attr('text-anchor', 'middle');
+  }
+
+  const bbox = textElem.node().getBBox();
 
-  const txt = textElem
-    .append('text')
-    .text(node.descr)
-    .attr('dy', '1em')
-    .attr('alignment-baseline', 'middle')
-    .attr('dominant-baseline', 'middle')
-    .attr('text-anchor', 'middle')
-    .call(wrap, node.width);
-  const bbox = txt.node().getBBox();
   const fontSize =
     conf.fontSize && conf.fontSize.replace ? conf.fontSize.replace('px', '') : conf.fontSize;
   node.height = bbox.height + fontSize * 1.1 * 0.5 + node.padding;
   node.height = Math.max(node.height, node.maxHeight);
   node.width = node.width + 2 * node.padding;
 
-  textElem.attr('transform', 'translate(' + node.width / 2 + ', ' + node.padding / 2 + ')');
+  if (!htmlLabels) {
+    const dx = node.width / 2;
+    const dy = node.padding / 2;
+    textElem.attr('transform', 'translate(' + dx + ', ' + dy + ')');
+  } else {
+    const dx = (node.width - bbox.width) / 2;
+    const dy = (node.height - bbox.height) / 2;
+    textElem.attr('transform', 'translate(' + dx + ', ' + dy + ')');
+  }
```

Comment:
No, it cannot. Scope of the variables is inside `if` statement only. They must be not `const` but `var` then and be declared before `if` statement.

Reply:
I see, I thought that if we have all possible cases then it would recognize it as an already declared variable. In this case I'm not sure if we should modify it to this:

```ts
let dx: number, dy: number;
if (!htmlLabels) {
   dx = node.width / 2;
   dy = node.padding / 2;
 } else {
   dx = (node.width - bbox.width) / 2;
   dy = (node.height - bbox.height) / 2;
 }
textElem.attr('transform', `translate(${dx}, ${dy})`);
```

Reply:
The number of lines is same in both cases, and there are two more variables in the outer scope. 
So I don't think the current implementation should definitely be changed. 


If making things compact is a goal, we could do something like this. 
```ts
   const dx = (node.width - (htmlLabels ? bbox.width : 0)) / 2;
   const dy = (node.height - (htmlLabels ? bbox.height : 0)) / 2;
   textElem.attr('transform', `translate(${dx}, ${dy})`);
```

Reply:
I'll tidy this up - I'm can't remember why we don't include the bounding box for svgs - it may have just been attempting to make it consistent with before. I'll work this out first and then either simplify or use as per @sidharthv96 suggested code.

Reply:
And I'll add the documentation / additional tests as per top of conversation.